pandas/core/reshape/pivot.py

```diff
@@ -379,7 +379,7 @@ def _generate_marginal_results_without_values(
 ):
     if len(cols) > 0:
         # need to "interleave" the margins
-        margin_keys = []
+        margin_keys: Union[List, Index] = []
```

Comment:
could be ArrayLike? (not sure it matters, but can update in followon if this works)

Reply:
That would work, because ``Union[List, Index]`` is the precise type of this.